plugins/woocommerce/src/Admin/API/OnboardingPlugins.php

```diff
@@ -411,4 +413,41 @@ public function get_install_activate_schema() {
 			),
 		);
 	}
+
+	public function log_plugins_install_error( $slug, $api, $result, $upgrader ) {
+		$properties = array(
+			'error_message'         => sprintf(
+			/* translators: %s: plugin slug (example: woocommerce-services) */
+				__(
+					'The requested plugin `%s` could not be installed.',
+					'woocommerce'
+				),
+				$slug
+			),
+			'type'				    => 'plugin_info_api_error',
+			'slug'                  => $slug,
+			'api_version'           => $api->version,
+			'api_download_link'     => $api->download_link,
+			'upgrader_skin_message' => implode( ',', $upgrader->skin->get_upgrade_messages() ),
+			'result'                => is_wp_error( $result ) ? $result->get_error_message() : 'null',
+		);
+		wc_admin_record_tracks_event( 'coreprofiler_install_plugin_error', $properties );
+	}
+
+	public function log_plugins_install_api_error( $slug, $api ) {
+		$properties = array(
+			'error_message'     => sprintf(
+			// translators: %s: plugin slug (example: woocommerce-services).
+				__(
+					'The requested plugin `%s` could not be installed. Plugin API call failed.',
+					'woocommerce'
+				),
+				$slug
+			),
+			'type'              => 'plugin_install_error',
+			'api_error_message' => $api->get_error_message(),
```

Comment:
I'm not sure if tracks value has any limit number of character, but I assume this can potentially get pretty long. I'll investigate if it's going to be a problem. Just flagging it for now

Reply:
Based on p1688977040450599-slack-C03DKP1JP, it seems we don't have a limit for tracks, but rather the bottleneck is on HTTP/GET request size, which is around 2kb. We should be safe 👍

Reply:
Thank you for checking on this @ilyasfoo 🙇 TIL